### PR TITLE
Adds section about the different options to use NLP in the stack

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

Related issue: https://github.com/elastic/search-docs-team/issues/83
This PR expands the NLP overview section with the _NLP in the Elastic stack_ subsection. This text explains the different options the users have for using NLP:
* NLP models as a service through the Inference API,
* NLP models uploaded and deployed in the cluster,
* store embeddings in an ES vector database.

### Preview

[NLP in the Elastic stack](https://stack-docs_bk_2679.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-nlp-overview.html#nlp-elastic-stack)

### CODE REVIEW

File: docs/en/stack/ml/nlp/ml-nlp-overview.asciidoc [hunk 1]
```diff
@@ -4,6 +4,34 @@
...
+You can **integrate NLP models from different providers** such as Cohere,
+HuggingFace, or OpenAI and use them as a service through the 
+{ref}/inference-apis.html[{infer} API]. You can also use <<ml-nlp-elser,ELSER>>
+(the retrieval model trained by Elastic) and <<ml-nlp-e5,E5>> in the same way.
+This {ref}/semantic-search-inference.html[tutorial] walks you through the
+process of using the various services with the {infer} API.
+
+You can **upload and manage NLP models** using the Eland client and the
+<<ml-nlp-deploy-models,{stack}>>. Find the 
+<<ml-nlp-model-ref,list of supported architectures here>>. Refer to <<setup>> to
```
Comment (davidkyle) [inline docs/en/stack/ml/nlp/ml-nlp-overview.asciidoc:23]:
```suggestion
<<ml-nlp-model-ref,list of recommended and compatible models here>>. Refer to <<setup>> to
```

Comment (davidkyle) [inline docs/en/stack/ml/nlp/ml-nlp-overview.asciidoc:23]: Do you want to have a link to `Setup and Security` here in the introduction? What it links to is quite low level details.

Perhaps link to the examples at https://stack-docs_bk_2679.docs-preview.app.elstc.co/guide/en/machine-learning/master/ml-nlp-examples.html

**Comment by szabosteve** [comment]
@davidkyle Thanks for your comments, I addressed them via [2ce1a67](https://github.com/elastic/stack-docs/pull/2679/commits/2ce1a6754a82e1da8f83f7cc2de292478876ca39).

**Review by davidkyle — APPROVED** [review]
LGTM

**Pull request merged**